web/src/components/IndexationForm.vue

```diff
@@ -0,0 +1,220 @@
+      </template>
+    </form>
+</template>
+
+<script>
+import Datepicker from '@vuepic/vue-datepicker';
+import '@vuepic/vue-datepicker/dist/main.css';
+
+export default {
+  name: 'IndexationForm',
+  data () {
+    return {
+      submitting: false,
+      formParams: {
+        baseRent: '',
+        region: '',
+        signedOn: '',
+        startDate: '',
+      },
+      activeRadio: ''
+    };
+  },
+  components: {
+    Datepicker
+  },
+  computed: {
+    missingBaseRent() {
+      return this.formParams.baseRent === '';
+    },
+    missingRegion() {
+      return this.formParams.region === '';
+    },
+    missingSignedOn() {
+      return this.formParams.signedOn === '';
+    },
+    missingStartDate() {
+      return this.formParams.startDate === '';
+    },
+    startDateIsInFuture() {
+      return this.formParams.startDate > new Date();
+    },
+    signedOnIsInFuture() {
+      return this.formParams.signedOn > new Date();
+    },
+    startBeforeSign() {
+      return this.formParams.startDate < this.formParams.signedOn;
+    },
+    negativeOrZeroBaseRent() {
+      return parseInt(this.formParams.baseRent) <= 0;
+    },
+    formIsInvalid() {
+      return this.missingBaseRent        ||
+             this.missingRegion          ||
+             this.missingSignedOn        ||
+             this.missingStartDate       ||
+             this.startDateIsInFuture    ||
+             this.signedOnIsInFuture     ||
+             this.startBeforeSign        ||
+             this.negativeOrZeroBaseRent;
+    },
+    errorClassForBaseRent() {
+      return (this.submitting && (this.missingBaseRent || this.negativeOrZeroBaseRent)) ? 'invalid-input' : '';
+    },
+    errorClassForSignedOn() {
+      return (this.submitting && (this.missingSignedOn || this.signedOnIsInFuture || this.startBeforeSign)) ? 'invalid-input' : '';
+    },
+    errorClassForStartDate() {
+      return (this.submitting && (this.missingStartDate || this.startDateIsInFuture || this.startBeforeSign)) ? 'invalid-input' : '';
+    }
+  },
+  methods: {
+    trySubmit() {
+      this.submitting = true
+      if (this.formIsInvalid) {
+        return;
+      }
+      this.convertNonStringsToRightType();
+      this.formParams = this.formatParamsForCall(this.formParams);
+      this.$emit('give-valid-params-for-call:formParams', this.formParams);
+      this.submitting = false;
+    },
+    convertNonStringsToRightType() {
+      this.formParams.baseRent = parseInt(this.formParams.baseRent);
+      this.formParams.signedOn = this.formParams.signedOn.toISOString().split('T')[0];
+      this.formParams.startDate = this.formParams.startDate.toISOString().split('T')[0];
+    },
+    formatParamsForCall(formParams) {
+      return {
+        'base_rent': formParams.baseRent,
+        'region': formParams.region.toLowerCase(),
+        'signed_on': formParams.signedOn,
+        'start_date': formParams.startDate
+      };
+    },
```

Comment:
Some comments here. While it's not "bad" (it works 😄), there are some things that I think shouldn't be done that way.

1. **Separation of concerns**: You have all the logic about the form in this file (state, validations, submit method), yet you have the actual submit call that's done in `App.vue`. I would have done everything here and just emitted the result to the parent component
```js
handleSubmit () {
  // xxx
  const response = fetch('xxxx', params)
  this.$emit('results', response)
}
```
2. **Handling loading states**: You're doing kind of a loading state with `this.submitting = true` but since there is no async operations in the method, you can't really show a loading state. If you make the call in the same method as in point 1, you can then handle the loading state. You can then also prevent the user from submitting the data a second time.
```js
// You can show a loading spinner if `isSubmitting` is true
async handleSubmit () {
  if (this.formIsInvalid || this.isSubmitting) return // Prevent submitting the form if it's invalid or if it's already submitting
  this.isSubmitting = true
  const response = await fetch(xxx)
  this.isSubmitting = false
}
```
3. **Mutating the `formParams` object**: You're kind of messing with the data here. You have an initial `formParams` object that has a specific structure `formParams.baseRent` for example. When you're submitting, you're overriding that initial object with a new object with different keys `formParams.base_rent`. So everything that's relying on that initial object is "broken" (your computed properties, your form in the HTML. What I would have done is keeping the state untouched, and serializing the data just for the call.
```js
async handle submit () {
  // ...
  const response = await fetch('xxx', this.serializeParams())
  // ...
},

serializeParams () {
  return {
    base_rent: parseInt(this.formParams.baseRent),
    region: this.formParams.region, // If your values are correctly set, no need for toLowerCase()
    signed_on: this.formParams.signedOn.toISOString().split('T')[0],
    start_date: this.formParams.startDate.toISOString().split('T')[0]
  }
}
```